core/src/main/java/org/ehcache/core/EhcacheManager.java

```diff
@@ -63,76 +110,548 @@ public EhcacheManager(Configuration config, Collection<Service> services) {
+    Set<ResourceType> resources = config.getResourcePools().getResourceTypeSet();
+    ServiceConfiguration<?>[] serviceConfigArray = serviceConfigs.toArray(new ServiceConfiguration[serviceConfigs.size()]);
+    if (serialization != null) {
+      try {
+        final Serializer<K> keySer = serialization.createKeySerializer(keyType, config.getClassLoader(), serviceConfigArray);
+        lifeCycledList.add(new LifeCycledAdapter() {
+          @Override
+          public void close() throws Exception {
+            serialization.releaseSerializer(keySer);
+          }
+        });
+        keySerializer = keySer;
+      } catch (UnsupportedTypeException e) {
+        for (ResourceType resource : resources) {
+          if (resource.requiresSerialization()) {
+            throw new RuntimeException(e);
+          }
+        }
+        this.getLogger().debug("Could not create serializers for " + alias, e);
+      }
       try {
-        destroyPersistenceSpace(alias);
-      } catch (CachePersistenceException e) {
-        this.getLogger().warn("Unable to clear persistence space for cache {}", alias, e);
+        final Serializer<V> valueSer = serialization.createValueSerializer(valueType, config.getClassLoader(), serviceConfigArray);
+        lifeCycledList.add(new LifeCycledAdapter() {
+          @Override
+          public void close() throws Exception {
+            serialization.releaseSerializer(valueSer);
+          }
+        });
+        valueSerializer = valueSer;
+      } catch (UnsupportedTypeException e) {
+        for (ResourceType resource : resources) {
+          if (resource.requiresSerialization()) {
+            throw new RuntimeException(e);
+          }
+        }
+        this.getLogger().debug("Could not create serializers for " + alias, e);
+      }
+    }
+
+    int eventParallelism;
+    StoreEventSourceConfiguration eventSourceConfiguration = ServiceLocator.findSingletonAmongst(StoreEventSourceConfiguration.class, config
+        .getServiceConfigurations()
+        .toArray());
+    if (eventSourceConfiguration != null) {
+      eventParallelism = eventSourceConfiguration.getOrderedEventParallelism();
+    } else {
+      eventParallelism = StoreEventSourceConfiguration.DEFAULT_EVENT_PARALLELISM;
+    }
+
+    Store.Configuration<K, V> storeConfiguration = new StoreConfigurationImpl<K, V>(config, eventParallelism, keySerializer, valueSerializer);
+    final Store<K, V> store = storeProvider.createStore(storeConfiguration, serviceConfigArray);
+
+    lifeCycledList.add(new LifeCycled() {
+      @Override
+      public void init() throws Exception {
+        storeProvider.initStore(store);
+      }
+
+      @Override
+      public void close() {
+        storeProvider.releaseStore(store);
+      }
+    });
+
+    return store;
+  }
+
+  private PersistableResourceService getPersistableResourceService(ResourceType resourceType) {
```

Comment:
After discussion with @cljohnso the `ResourceType` is enough to handle that future.